au/conversion_strategy.hh

```diff
@@ -68,31 +68,67 @@ struct ApplicationStrategyForImpl<T, Mag, MagKind::INTEGER_DIVIDE>
 template <typename T, typename Mag>
 struct ApplicationStrategyForImpl<T, Mag, MagKind::NONTRIVIAL_RATIONAL>
     : std::conditional<
-          std::is_integral<T>::value,
+          std::is_integral<RealPart<T>>::value,
           OpSequence<MultiplyTypeBy<T, NumeratorT<Mag>>, DivideTypeByInteger<T, DenominatorT<Mag>>>,
           MultiplyTypeBy<T, Mag>> {};
 
 //
-// `FullConversionImpl<OldRep, PromotedCommon, NewRep, Factor>` should resolve to the most efficient
+// `RepForConversion<OldRep, NewRep>` is the rep we should use when applying the conversion factor.
+//
+template <typename OldRep, typename NewRep>
+struct ApplicationRepImpl;
```

Comment:
"Application" could also imply "the application using this library."  Would `ConversionRep` be more clear in this case?  Or maybe `IntermediateRep`?

Reply:
Good point.  I went with `ConversionRep`.  Seems like a pretty clear improvement.